src/client/client.go

```diff
@@ -765,17 +782,14 @@ func (c *APIClient) connect(timeout time.Duration, unaryInterceptors []grpc.Unar
 	}
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
 	defer cancel()
-	addr := c.addr
-	if !strings.HasPrefix(addr, "dns:///") {
-		addr = "dns:///" + c.addr
-	}
 
-	// TODO: the 'dns:///' prefix above causes connecting to hang on windows
-	// unless we also prevent the resolver from fetching a service config (which
-	// we don't use anyway).  Don't ask me why.
+	// By default GRPC will attempt to get service config from a TXT record when
+	// the `dns:///` scheme is used. Some DNS servers return the wrong type of error
+	// when the TXT record doesn't exist, and it causes GRPC to back off and retry
+	// service discovery forever.
```

Comment:
Mystery solved!

Reply:
Yeah we solved this mystery like 6 months ago when a non-windows user had the same problem connecting to their Hub cluster, but I never went back and updated this comment.